File: test_api.js
const fs = require('fs');

async function test() {
  fs.writeFileSync('dummy.mov', 'ID3\x03\x00\x00\x00\x00\x00\x00');
  
  const blob = new Blob([fs.readFileSync('dummy.mov')], { type: 'video/quicktime' });
  const formData = new FormData();
  formData.append('file', blob, 'dummy.mov');
  formData.append('voiceId', '1');
  formData.append('emotion', '1');

  try {
    const res = await fetch('http://localhost:8080/api/video-editor/voice-swap', {
      method: 'POST',
      body: formData
    });
    const data = await res.text();
    console.log("STATUS:", res.status);
    console.log("RESPONSE:", data);
  } catch (err) {
    console.error("ERROR:", err);
  }
}

test();

File: test_api.py
import requests

url = "http://localhost:8080/api/video-editor/voice-swap"

# Create a dummy audio file
with open("dummy.mp3", "wb") as f:
    f.write(b"ID3\x03\x00\x00\x00\x00\x00\x00")

with open("dummy.mp3", "rb") as f:
    files = {"file": ("dummy.mp3", f, "audio/mpeg")}
    data = {"voiceId": "1", "emotion": "1"}
    response = requests.post(url, files=files, data=data)

print(response.status_code)
print(response.text)
